### PR TITLE
Sanitize queries we send to Freelex

### PR DESCRIPTION
**Pull request by eoinkelly**
Fixes issue #978 

This implements sanitization of **all** the query parameters we receive before sending them to Freelex. The rules for sanitization vary by parameter but hopefully the logic can be understood from the code.

We should test this manually when it hits staging to be sure that it hasn't changed any of the queries.

### CODE REVIEW

**Comment by eoinkelly** [comment]
Despite Codeclimate's feelings on the matter, I stand by my layout of `sanitize_for_standard_search` - I think breaking it up would actually make it harder to discern the pattern in the (fairly complex) business logic

File: app/services/search_query_sanitization_service.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  ##
+  # @param [String] handshape
+  # @return [String]
+  #
+  def sanitize_handshape(handshape)
+    return '' if handshape.nil?
+
+    handshape
+      .gsub(ANYTHING_EXCEPT_NUMBERS_SPACE_PERIOD_REGEX, '')
```
Comment (Br3nda) [inline app/services/search_query_sanitization_service.rb:66]: the long names did make me smile, but they are good names

Comment (Br3nda) [reply app/services/search_query_sanitization_service.rb:66]: (but btw in NZ we call that a fullstop, not a period, afaik)

Comment (eoinkelly) [reply app/services/search_query_sanitization_service.rb:66]: Same in Ireland but that was even longer! 😀

**Pull request merged**